### ISSUE TITLE
Please add custom description support for code actions.

### ISSUE THREAD

**Issue by Weiged** [carlrobertoh/CodeGPT#20]
Native speakers of languages other than English need to add custom descriptions.

**Comment by carlrobertoh** [comment]
Hi @Weiged 

Thank you for the feature request. I'll see what I can do.

Best regards

**Comment by rangzen** [comment]
Something like a "Do not send automatically the request" in the settings will let you edit the request before sending it.
You can also add a field "language (speaking)" with default in English and add something like a postfix at each request ", and respond in %s" with the language...
It is some possible workaround waiting for something more complex like a table with the menu entries with Display name and editable prompts.

**Comment by carlrobertoh** [comment]
I'm not entirely sure how people configure their IDE language, is it some kind of external plugin that needs to be installed? If yes, then perhaps I might be able to build some sort of a language mapping. I have seen it is possible from some screenshots, but haven't dig deeper.

> I have in mind a "language (code)" setting with an option "auto from IDE" to force the answer to a specific language or tool (docs, SQL, schemas like mermaid, etc.)

That's really nice idea and actually I already thought about it when implementing the Custom Prompt text area.
https://github.com/carlrobertoh/CodeGPT/blob/master/src/main/java/ee/carlrobert/chatgpt/ide/action/CustomPromptDialog.java#L49
Basically it will extract the file extension where the selected code originates from and then maps to the correct language/syntax.

**Comment by rangzen** [comment]
I think this type of setting will resolve a lot of feature requests about translation, customs, etc.
This is from the running configuration, but it is to show what I have in mind.
The "Name" will be the names in the CodeGPT submenu.

![image](https://user-images.githubusercontent.com/132700/223470512-ed230e2e-673d-4158-8f4c-9499df3b9a5f.png)

**Comment by rangzen** [comment]
As for the language, I think it is more subtle and come from this
![image](https://user-images.githubusercontent.com/132700/223471535-55a75825-b453-42e5-bfe8-0ccbb973a8f2.png)

**Comment by rangzen** [comment]
Oups, theres is a grammar mistake in the French example about Custom Prompts :D

**Comment by carlrobertoh** [comment]
Solving the custom prompts via env variables is actually really clever idea, altho it doesn't necessarily have to be envs, could be just an application level storage/state. I think this deserves a separate feature ticket and requires bit of a work.

Regarding to the language, I'm not sure if I understood you correctly. The problem with "auto from IDE" is that the selected code can be different language(same project, different file/language), so I'm not sure if it will work correctly.

**Comment by rangzen** [comment]
Yeah, envvars was just to show you the type of GUI I have in mind: a table with +, -, copy, etc. But yes, it should be in the settings. It will be easier to have a name to use in the menu instead of envvars with names like CODEGPT_CUSTOM_1, CODEGPT_CUSTOM_2, or anything like that. And people are already confused just for the Open AI key so I don't imagine the requests from Windows users if something like that end up in envvars...
I will open a new issue to discuss about it.

For the language, I think Jetbrains use the File types and some checks. You right, it can give a hint but you cannot rely on it if you select SQL from a Go file or mermaid from Javascript comments, etc.

**Comment by Weiged** [comment]
I tried CustomPrompt and found it not what I wanted. What I mean is that the predefined actions such as "Find Bugs" support custom descriptions.

**Comment by rangzen** [comment]
@Weiged, this is what we are talking about, how to implement this type of feature.

**Comment by carlrobertoh** [comment]
Hi, the support for custom actions has been added in version 1.5.0.

<img width="1031" alt="Screen Shot 2023-03-15 at 10 57 01" src="https://user-images.githubusercontent.com/15891408/225290084-7455bc37-b9e9-41b1-9a93-88526d691b3c.png">
<img width="965" alt="Screen Shot 2023-03-15 at 10 56 46" src="https://user-images.githubusercontent.com/15891408/225290091-c465ada9-f1fa-4945-bea2-97694d76b289.png">

I hope that's what you were expecting.

Best regards
